cmp/chart/impl/ChartContextMenuItems.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2025 Extremely Heavy Industries Inc.
+ */
+import {isValidElement, MouseEvent, ReactNode} from 'react';
+import {cloneDeep, isEmpty, isString} from 'lodash';
+import {ChartModel} from '@xh/hoist/cmp/chart';
+import {MenuItem} from '@xh/hoist/core';
+import {Highcharts} from '@xh/hoist/kit/highcharts';
+import {Icon} from '@xh/hoist/icon';
+
+/**
+ * Highcharts supported tokens {@link https://api.highcharts.com/highcharts/exporting.buttons.contextButton.menuItems}
+ * plus Hoist's `copyToClipboard`.
+ */
+export type ChartContextMenuToken =
+    | 'viewFullscreen'
+    | 'printChart'
+    | 'downloadJPEG'
+    | 'downloadPNG'
+    | 'downloadSVG'
+    | 'downloadCSV'
+    | 'downloadXLS'
+    | 'downloadPDF'
+    | 'copyToClipboard';
+
+export interface ChartMenuItem extends Omit<MenuItem, 'actionFn' | 'items'> {
+    items?: ChartMenuItemLike[];
+    actionFn?: (
+        menuItemEvent: MouseEvent | PointerEvent,
+        contextMenuEvent: MouseEvent | PointerEvent,
+        params: {
+            chartModel: ChartModel;
+            /**
+             * Single point is the active series point the mouse is closest to
+             */
+            point: any;
+            /**
+             * Points array is the list of points hovered over in each series. When
+             * there are multiple series and tooltip.shared = true, points.length > 1.
+             */
+            points: any[];
+        }
+    ) => void;
+}
+
+/**
+ * If a String, value can be '-' for a separator, or a token supported by HighCharts
+ * for its native menu items, or a Hoist specific token.
```

Comment:
Can it also be any arbitrary string to insert a header?  If not, think we should spec `'-'` vs `string` here